### PR TITLE
CV_8UC1 in realsense

### PR DESCRIPTION
**Pull request by cabelo**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [X] The PR is proposed to proper branch
- [X] There is reference to original bug report and related work
- [X] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [X] The feature is well documented and sample code can be built with the project CMake

Evidence about this PR with realsense D435 :
image

image

### CODE REVIEW

File: doc/tutorials/app/intelperc.markdown [hunk 1]
```diff
@@ -24,13 +24,13 @@ In order to use depth sensor with OpenCV you should do the following preliminary
 VideoCapture can retrieve the following data:
 
 -#  data given from depth generator:
-    -   CAP_INTELPERC_DEPTH_MAP - each pixel is a 16-bit integer. The value indicates the
-            distance from an object to the camera's XY plane or the Cartesian depth. (CV_16UC1)
+    -   CAP_INTELPERC_DEPTH_MAP - each pixel is a 16-bit or 8-bit integer. The value indicates the
+            distance from an object to the camera's XY plane or the Cartesian depth. (CV_16UC1 or CV_8UC1)
```
Comment (dkurt) [inline doc/tutorials/app/intelperc.markdown:28]: How the following code works this way?

https://github.com/opencv/opencv/blob/fc1a15626226609babd128e043cf7c4e32f567ca/modules/videoio/src/cap_librealsense.cpp#L199-L202

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown:28]: Hi, @dkurt 

To work well, I had to make this code in D435:
    Mat depthMap;
    Mat image;
    Mat irImage;
    Mat gray;

    capture.grab();
    capture.retrieve( depthMap, CAP_INTELPERC_DEPTH_MAP );
    capture.retrieve(    image, CAP_INTELPERC_IMAGE );
    capture.retrieve(  irImage, CAP_INTELPERC_IR_MAP);

    depthMap.convertTo(gray,CV_8U);
    
    imshow("Live 1", gray);
    imshow("Live 2", image);
    imshow("Live 3", irImage);

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown:28]: image

Mat depthMap;
Mat image;
Mat irImage;
Mat gray;
capture.grab();
capture.retrieve( depthMap, CAP_INTELPERC_DEPTH_MAP );
capture.retrieve(    image, CAP_INTELPERC_IMAGE );
capture.retrieve(  irImage, CAP_INTELPERC_IR_MAP);
imshow("Live 1", depthMap);
imshow("Live 2", image);
imshow("Live 3", irImage);

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown:28]: Hi, @dkurt,

Solved see!

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown]: https://github.com/opencv/opencv/pull/19231/commits/96bd31fa9b7df3e89e0683840700035779d588d6

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown]: modules/videoio/src/cap_librealsense.cpp

Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown]: https://github.com/opencv/opencv/blob/96bd31fa9b7df3e89e0683840700035779d588d6/modules/videoio/src/cap_librealsense.cpp#L199-L201

File: modules/videoio/src/cap_librealsense.hpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ class VideoCapture_LibRealsense : public IVideoCapture
     rs2::pipeline mPipe;
     rs2::frameset mData;
     rs2::align    mAlign;
+    rs2::colorizer color_map;
```
Comment (paroj) [inline modules/videoio/src/cap_librealsense.hpp:33]: see `cv::applyColorMap` for doing this without touching the capture

File: doc/tutorials/app/intelperc.markdown (hunk 1 above)
Comment (alalek) [reply doc/tutorials/app/intelperc.markdown]: > depthMap.convertTo(gray,CV_8U);

Wrong way.
In your proposal you can even declare support for 64F with the same success.

File: modules/videoio/src/cap_librealsense.cpp [hunk 3]
```diff
@@ -197,8 +197,8 @@ bool VideoCapture_LibRealsense::retrieveFrame(int outputType, cv::OutputArray fr
     switch (outputType)
     {
     case CAP_INTELPERC_DEPTH_MAP:
-        _frame = mData.get_depth_frame().as<rs2::video_frame>();
-        type = CV_16UC1;
+        _frame = mData.get_depth_frame().apply_filter(color_map);
+        type = CV_8UC3;
```
Comment (alalek) [inline modules/videoio/src/cap_librealsense.cpp:201]: Broken compatibility.

Closing PR as "invalid".

File: doc/tutorials/app/intelperc.markdown (hunk 1 above)
Comment (cabelo) [reply doc/tutorials/app/intelperc.markdown]: ok, thanks

**Pull request closed without merging**